test/erlfmt_format_SUITE.erl

```diff
@@ -545,6 +547,200 @@ binary_operator(Config) when is_list(Config) ->
+        "        register_login(User, dates:now()),\n"
+        "        [open_admin_session(User) || is_admin(User)]\n"
+        "    end\n"
+    ),
+    ?assertSame(
+        "{B, C} =\n"
+        "    case X of\n"
+        "        true -> f();\n"
+        "        _ -> g()\n"
+        "    end\n"
+    ),
+    ?assertSame(
+        "A = {B, C} =\n"
+        "    case X of\n"
+        "        true -> f();\n"
+        "        _ -> g()\n"
+        "    end.\n"
+    ),
+    ?assertSame(
+        "A = {{B1, B2}, C} = {B, C} =\n"
+        "    case X of\n"
+        "        true -> f();\n"
+        "        _ -> g()\n"
+        "    end.\n"
+    ),
+    ?assertSame(
+        "A =\n"
+        "    ({{B1, B2}, C} =\n"
+        "        ({B, C} =\n"
+        "            case X of\n"
+        "                true -> f();\n"
+        "                _ -> g()\n"
+        "            end)).\n"
+    ),
+    ?assertSame(
+        "convert_meta(Key, Value) when\n"
+        "    is_list(Key) orelse is_binary(Key),\n"
+        "    is_list(Key) orelse\n"
+        "        is_binary(Value) orelse\n"
+        "        is_atom(Value) orelse\n"
+        "        is_integer(Value)\n"
+        "->\n"
+        "    ok.\n"
+    ),
+    ?assertSame(
+        "equivalent(L1, R1) andalso equivalent(L2, R2) andalso equivalent(L3, R3) andalso\n"
+        "    equivalent(L4, R4)\n"
+    ),
+    ?assertSame(
+        "equivalent(L1, R1) andalso\n"
+        "    equivalent(L2, R2) andalso equivalent(L3, R3) andalso equivalent(L4, R4)\n"
+    ),
+    ?assertFormat(
+        "equivalent(L1, R1) andalso equivalent(L2, R2) andalso equivalent(L3, R3) andalso equivalent(L4, R4)\n",
+        "equivalent(L1, R1) andalso equivalent(L2, R2) andalso equivalent(L3, R3) andalso\n"
+        "    equivalent(L4, R4)\n"
+    ),
+    ?assertFormat(
+        "the:ever(funny) andalso begin blocksof:my(code), that:appear(between), your:boolean(expressions) end andalso are:terrible(things, to, debug)\n",
+        "the:ever(funny) andalso\n"
+        "    begin\n"
+        "        blocksof:my(code),\n"
+        "        that:appear(between),\n"
+        "        your:boolean(expressions)\n"
+        "    end andalso are:terrible(things, to, debug)\n"
+    ),
+    ?assertFormat(
+        "a_function:with_a(<<\"very\">>, Long, list, \"of\", <<\"attributes\">>) andalso a:short(one) andalso another:very_very_long_one() andalso Another\n",
+        "a_function:with_a(<<\"very\">>, Long, list, \"of\", <<\"attributes\">>) andalso\n"
+        "    a:short(one) andalso another:very_very_long_one() andalso Another\n"
+    ),
+    ?assertSame(
+        "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA orelse\n"
+        "    BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB andalso\n"
+        "        CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC.\n"
+    ),
+    ?assertSame(
+        "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa andalso\n"
+        "    bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb orelse\n"
+        "    cccccccccccccccccccccccccccccccccc.\n"
```

Comment:
I understand why it's formatted like this, but it's not very clear when just taking a brief look. I wish we could add parens in cases like this, it would make it much, much clearer:
```erlang
(aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa andalso
    bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb) orelse
    cccccccccccccccccccccccccccccccccc
```

Reply:
I can add the test with parens, but other than that I think it is important to also test without parens

Reply:
I had added the extra test, but don't know what else to do